web/components/bet/quick-bet.tsx

```diff
@@ -150,57 +154,137 @@ export function QuickBet(props: {
       <Row
         className={clsx(
           className,
-          'absolute my-auto mt-0.5 w-full items-center justify-between align-middle'
+          'absolute my-auto mt-1 w-full items-center justify-between px-2 align-middle'
         )}
       >
-        <Row
-          className="items-center pr-4"
+        <BinaryQuickBetButton
+          onClick={() => placeQuickBet('DOWN')}
+          direction="DOWN"
+          user={user}
+          contract={contract}
+          hover={downHover}
           onMouseEnter={() => setDownHover(true)}
           onMouseLeave={() => setDownHover(false)}
-          onClick={() => placeQuickBet('DOWN')}
-        >
-          <TriangleLeftFillIcon
-            className={clsx(
-              'mx-auto h-6 w-6',
-              downHover ? 'text-indigo-700' : 'text-indigo-500'
-            )}
-          />
-          <span
-            className={clsx(
-              'text-sm font-light text-indigo-500 transition-opacity',
-              downHover ? 'opacity-100' : 'opacity-0 '
-            )}
-          >
-            M$10
-          </span>
-        </Row>
-        <Row
-          className="items-center pl-4"
+        />
+        <BinaryQuickBetButton
+          onClick={() => placeQuickBet('UP')}
+          direction="UP"
+          user={user}
+          contract={contract}
+          hover={upHover}
           onMouseEnter={() => setUpHover(true)}
           onMouseLeave={() => setUpHover(false)}
-          onClick={() => placeQuickBet('UP')}
-        >
-          <span
-            className={clsx(
-              'transition- text-sm font-light text-indigo-500',
-              upHover ? 'opacity-100' : 'opacity-0 '
-            )}
-          >
-            M$10
-          </span>
-          <TriangleRightFillIcon
-            className={clsx(
-              'mx-auto h-6 w-6',
-              upHover ? 'text-indigo-700' : 'text-indigo-500'
-            )}
-          />
-        </Row>
+        />
       </Row>
       <QuickOutcomeView contract={contract} previewProb={previewProb} />
     </div>
   )
 }
 
+function BinaryQuickBetButton(props: {
+  onClick: () => void
+  direction: 'UP' | 'DOWN'
+  user: User
+  contract: Contract
+  hover: boolean
+  onMouseEnter: () => void
+  onMouseLeave: () => void
+}) {
+  const {
+    onClick,
+    direction,
+    user,
+    contract,
+    hover,
+    onMouseEnter,
+    onMouseLeave,
+  } = props
+  const userBets = useUserContractBets(user.id, contract.id)
```

Comment:
This hook unfortunately fetches new data every time.

Maybe just pass in `userBets` since the parent has it.